airbyte-cdk/python/unit_tests/sources/declarative/auth/test_requests_native_auth.py

```diff
@@ -0,0 +1,160 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+
+import pendulum
+import pytest
+import requests
+from airbyte_cdk.sources.declarative.auth.oauth import Oauth2Authenticator
+from requests import Response
+
+LOGGER = logging.getLogger(__name__)
+
+resp = Response()
+
+config = {
+    "refresh_endpoint": "refresh_end",
+    "client_id": "some_client_id",
+    "client_secret": "some_client_secret",
+    "refresh_token": "some_refresh_token",
+    "token_expiry_date": pendulum.now().subtract(days=2).to_rfc3339_string(),
+    "custom_field": "in_outbound_request",
+    "another_field": "exists_in_body",
+}
+
+
+class TestOauth2Authenticator:
```

Comment:
This is a rough copy of the existing unit tests to ensure we retain the same feature set, but w/ a few extra test variations that rely on inputs that require interpolation.